test/framework/src/main/java/org/elasticsearch/transport/nio/SocketEventHandler.java

```diff
@@ -143,8 +144,8 @@ public void writeException(NioSocketChannel channel, Exception exception) {
      * @param channel that caused the exception
      * @param exception that was thrown
      */
-    public void genericChannelException(NioSocketChannel channel, Exception exception) {
-        logger.trace("event handling failed", exception);
+    void genericChannelException(NioSocketChannel channel, Exception exception) {
+        logger.debug(new ParameterizedMessage("exception while handling event for socket channel: {}", channel), exception);
```

Comment:
Since this is at the debug level, should you use a supplier here to avoid instantiating the parameterized message if not needed?